.github/workflows/auto-changeset.yml

```diff
@@ -0,0 +1,85 @@
+name: Changeset Automation
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
+      - ready_for_review
+
+jobs:
+  auto-changeset:
+    if: |
+      startsWith(github.repository, 'asyncapi/') &&
+      (!contains(github.event.pull_request.labels, 'skip-changeset')) &&
+      ( startsWith(github.event.pull_request.title, 'fix:') ||
+        startsWith(github.event.pull_request.title, 'feat:') ||
+        startsWith(github.event.pull_request.title, 'fix!:') ||
+        startsWith(github.event.pull_request.title, 'feat!:')
+      )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Checkout PR
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: gh pr checkout ${{ github.event.pull_request.number }}
+
+      - name: Install specific package in temp dir
```

Comment:
why is this kinda complex flow needed? why not just `npm install read-package-up@11.0.0` ?

Reply:
Two problems:
1. It installs everything in the package.json, don't want that.
2. It is problematic when used with studio or other monorepos due to workspaces.

Reply:
Happy to implement some alternative if you have some ideas?

Reply:
Added the reasons as discussed.